pkg/kubectl/cmd/exec.go

```diff
@@ -174,6 +174,12 @@ func (p *ExecOptions) Run() error {
 		return err
 	}
 
+	started := pod.Status.StartTime != nil
```

Comment:
Unfortunately, this is probably too strict.  You can actually have a pod up and running (reachable over the internet) and not be able to exec into it, because the apiserver has not been updated with the status.

We removed this check specifically because the Kubelet is capable of returning better messages.  Is there a reason you can't fix this in the kubelet?